cocos/2d/framework/canvas.ts

```diff
@@ -134,16 +134,33 @@ export class Canvas extends RenderRoot2D {
         this._thisOnCameraResized = this._onResizeCamera.bind(this);
 
         if (EDITOR) {
-            this._fitDesignResolution = (): void => {
+            this.fitDesignResolution_EDITOR = (): void => {
                 // TODO: support paddings of locked widget
                 this.node.getPosition(this._pos);
                 const nodeSize = view.getDesignResolutionSize();
-                Vec3.set(_worldPos, nodeSize.width * 0.5, nodeSize.height * 0.5, 0);
+                const trans = this.node._uiProps.uiTransformComp!;
+
+                let scaleX = this.node.scale.x;
```

Comment:
Don't and this one for now. Widgets adapting based on scale is a weird problem, and we're going to deprecate the effect of scale in widgets after this.